compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEAttrs.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEAttrs.h"
+
+#include "iree-amd-aie/IR/AMDAIEDialect.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Transform/IR/TransformOps.h"
+#include "mlir/IR/DialectImplementation.h"
+
+#define GET_ATTRDEF_CLASSES
+#include "iree-amd-aie/IR/AMDAIEAttrs.cpp.inc"
+#include "iree-amd-aie/IR/AMDAIEEnums.cpp.inc"
+
+static const char kPackingConfigAttrName[] = "packing_config";
+
+namespace mlir::iree_compiler {
+
+/// Returns an `ArrayAttr` where each element is an `IntegerAttr` of 64-bit
+/// integer type whose values is obtained from `values`.
+static ArrayAttr getIndexArrayAttr(MLIRContext *context,
+                                   ArrayRef<int64_t> values) {
+  return ArrayAttr::get(
+      context, llvm::map_to_vector(values, [&](int64_t value) -> Attribute {
+        return IntegerAttr::get(IndexType::get(context), APInt(64, value));
+      }));
+}
+
+}  // namespace mlir::iree_compiler
+
+namespace mlir::iree_compiler::AMDAIE {
+
+//===----------------------------------------------------------------------===//
+// amdaie.packing_config
+//===----------------------------------------------------------------------===//
+
+PackingConfigAttr PackingConfigAttr::get(
```

Comment:
If we drop the C++ struct, this back and forth isnt necessary. We can just use the attribute directly.